build.rs

```diff
@@ -11,6 +11,7 @@ fn main() {
         "cargo:rustc-env=RUST_HOST_TARGET={}",
         std::env::var("TARGET").unwrap()
     );
+    println!("cargo:rerun-if-changed=tests/snapshots");
```

Comment:
Talking this over with @ehuss, we are concerned about the balance we are striking between writing new tests and the cost of rebuilding.  In case its not clear, this line means that any time anything in the snapshots directory changes, whether directly by the user or from rebase from master, all of cargo, lib and bin, will rebuild.

If we had globbing support in `rerun-if-changed`, we could limit this to just `tests/snapshots/*/*/mod.rs` which might still cause too many rebuilds.

I know we proposed this solution and you put in a lot of work to learn how to write the function-like macro and get this working but I think we should change to a naive / macro-less solution for now where we rename `tests/testsuite/cargo_add.rs` to `tests/testsuite/cargo_add/mod.rs` and `tests/snapshots/add/add_basic/mod.rs` to `tests/testsuite/cargo_add/add_basic/mod.rs` and move the snapshots along with it so we can just `mod add_basic;` in `cargo_add/mod.rs`.  The downside is that test writers have to make sure they keep the linkage up-to-date and don't add tests without linking them in but we already have that problem in `tests/testsuite`.

Reply:
I completely agree with this decision. I do not think the rebuilds were worth it, even while testing this and making changes it was happening. It wasn't bad because I have a fast machine but it still slowed things down. It also completely mitigated some of the benefits of `snapshot`. 

I like the proposed solution better than using a macro here. Even if `rerun-if-changed` supported globs the explicitness of having to add a test might be nice. As for the downsides of having to remember to link everything up, for me personally, I don't think it's a problem since Clion says if a `mod.rs` isn't attached anywhere. That being said it may be nice to have something more set up so you don't have to manage paths in a `mod.rs` test

Reply:
> rename tests/testsuite/cargo_add.rs to tests/testsuite/cargo_add/mod.rs and tests/snapshots/add/add_basic/mod.rs to tests/testsuite/cargo_add/add_basic/mod.rs

Would this end up making more files with only one test case inside? I can imagine it is a little annoying when navigating a group of related tests but you need to jump back and forth between files.

Reply:
The whole aim of this direction is to have a file per test function, regardless of what the files are renamed.  I've been dealing with this in my other crates with `trycmd` and its not bad.  I also see a lot of this as an experiment.  We'll try things out, see what we think, and change as needed.